# Homework-3
# Bowler Career Analysis

# Name: Md. Rafsan Jamil
# ID: 20231056

# Objective:
1.To Find out the best bowler

2.To Find out the bowler who have the longest career

3.To Find out the bowler who have the lowest ecnomy

4.To Find out the bowler who have the lowest average

5.To Find out the bowler who have the lowest strike rate


In [38]:
#import libraries
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [39]:
df = pd.read_excel("ODI_cricket.xlsx",sheet_name='bowler')
df.head(10)

Player       Span  Mat  Inns  Balls   Runs  Wkts  \
0  M Muralitharan (Asia/ICC/SL)  1993-2011  350   341  18811  12326   534   
1             Wasim Akram (PAK)  1984-2003  356   351  18186  11812   502   
2            Waqar Younis (PAK)  1989-2003  262   258  12698   9919   416   
3          WPUJC Vaas (Asia/SL)  1994-2008  322   320  15775  11014   400   
4  Shahid Afridi (Asia/ICC/PAK)  1996-2015  398   372  17670  13632   395   
5       SM Pollock (Afr/ICC/SA)  1996-2008  303   297  15712   9631   393   
6          GD McGrath (AUS/ICC)  1993-2007  250   248  12970   8391   381   
7                   B Lee (AUS)  2000-2012  221   217  11185   8877   380   
8               SL Malinga (SL)  2004-2019  226   220  10936   9760   338   
9         A Kumble (Asia/INDIA)  1990-2007  271   265  14496  10412   337   

     Ave  Econ    SR   4   5  
0  23.08  3.93  35.2  15  10  
1  23.52  3.89  36.2  17   6  
2  23.84  4.68  30.5  14  13  
3  27.53  4.18  39.4   9   4  
4  34.51  4.62  44.7   4   9  
5  24.50  3.67  39.9  12   5  
6  22.02  3.88  34.0   9   7  
7  23.36  4.76  29.4  14   9  
8  28.87  5.35  32.3  11   8  
9  30.89  4.30  43.0   8   2

In [40]:
df.tail(3)

Player       Span  Mat  Inns  Balls  Runs  Wkts    Ave  Econ  \
74  Rashid Khan (AFG)  2015-2022   80    76   4074  2821   151  18.68  4.15   
75  UDU Chandana (SL)  1994-2007  147   136   6142  4818   151  31.90  4.70   
76   R Ashwin (INDIA)  2010-2022  113   111   6141  5058   151  33.49  4.94   

      SR  4  5  
74  26.9  5  4  
75  40.6  4  1  
76  40.6  1  0

In [41]:
df.columns

Index(['Player',   'Span',    'Mat',   'Inns',  'Balls',   'Runs',   'Wkts',
          'Ave',   'Econ',     'SR',        4,        5],
      dtype='object')

# Explanation of Each Column

Span= Range of his career, 

Mat= Matches, 

Inns=Innings,  

Wkts= Wickets,

Ave= Average, 

Econ= Economy, 

SR= Strike Rate, 

4= Four wicket achieve ,    

5= Five wicket achieve.

In [42]:
df.info

<bound method DataFrame.info of                           Player       Span  Mat  Inns  Balls   Runs  Wkts  \
0   M Muralitharan (Asia/ICC/SL)  1993-2011  350   341  18811  12326   534   
1              Wasim Akram (PAK)  1984-2003  356   351  18186  11812   502   
2             Waqar Younis (PAK)  1989-2003  262   258  12698   9919   416   
3           WPUJC Vaas (Asia/SL)  1994-2008  322   320  15775  11014   400   
4   Shahid Afridi (Asia/ICC/PAK)  1996-2015  398   372  17670  13632   395   
..                           ...        ...  ...   ...    ...    ...   ...   
72          SR Tendulkar (INDIA)  1989-2012  463   270   8054   6850   154   
73               BAW Mendis (SL)  2008-2015   87    84   4154   3324   152   
74             Rashid Khan (AFG)  2015-2022   80    76   4074   2821   151   
75             UDU Chandana (SL)  1994-2007  147   136   6142   4818   151   
76              R Ashwin (INDIA)  2010-2022  113   111   6141   5058   151   

      Ave  Econ    SR   4   5  

In [43]:
df.describe

<bound method NDFrame.describe of                           Player       Span  Mat  Inns  Balls   Runs  Wkts  \
0   M Muralitharan (Asia/ICC/SL)  1993-2011  350   341  18811  12326   534   
1              Wasim Akram (PAK)  1984-2003  356   351  18186  11812   502   
2             Waqar Younis (PAK)  1989-2003  262   258  12698   9919   416   
3           WPUJC Vaas (Asia/SL)  1994-2008  322   320  15775  11014   400   
4   Shahid Afridi (Asia/ICC/PAK)  1996-2015  398   372  17670  13632   395   
..                           ...        ...  ...   ...    ...    ...   ...   
72          SR Tendulkar (INDIA)  1989-2012  463   270   8054   6850   154   
73               BAW Mendis (SL)  2008-2015   87    84   4154   3324   152   
74             Rashid Khan (AFG)  2015-2022   80    76   4074   2821   151   
75             UDU Chandana (SL)  1994-2007  147   136   6142   4818   151   
76              R Ashwin (INDIA)  2010-2022  113   111   6141   5058   151   

      Ave  Econ    SR   4   5

In [44]:
df.shape

(77, 12)

In [45]:
df.isnull().sum()

Player    0
Span      0
Mat       0
Inns      0
Balls     0
Runs      0
Wkts      0
Ave       0
Econ      0
SR        0
4         0
5         0
dtype: int64

In [46]:
# renaming the column names
df = df.rename(columns={'Mat':'Match', 
                        'Inns':'Innings',
                        'Wkts': 'Wickets',
                        'SR': 'Strike Rate',
                        'Ave': 'Average',
                        'Econ': 'Economy'
                        })

In [47]:
df

Player       Span  Match  Innings  Balls   Runs  \
0   M Muralitharan (Asia/ICC/SL)  1993-2011    350      341  18811  12326   
1              Wasim Akram (PAK)  1984-2003    356      351  18186  11812   
2             Waqar Younis (PAK)  1989-2003    262      258  12698   9919   
3           WPUJC Vaas (Asia/SL)  1994-2008    322      320  15775  11014   
4   Shahid Afridi (Asia/ICC/PAK)  1996-2015    398      372  17670  13632   
..                           ...        ...    ...      ...    ...    ...   
72          SR Tendulkar (INDIA)  1989-2012    463      270   8054   6850   
73               BAW Mendis (SL)  2008-2015     87       84   4154   3324   
74             Rashid Khan (AFG)  2015-2022     80       76   4074   2821   
75             UDU Chandana (SL)  1994-2007    147      136   6142   4818   
76              R Ashwin (INDIA)  2010-2022    113      111   6141   5058   

    Wickets  Average  Economy  Strike Rate   4   5  
0       534    23.08     3.93         35.2  15  10  
1       502    23.52     3.89         36.2  17   6  
2       416    23.84     4.68         30.5  14  13  
3       400    27.53     4.18         39.4   9   4  
4       395    34.51     4.62         44.7   4   9  
..      ...      ...      ...          ...  ..  ..  
72      154    44.48     5.10         52.2   4   2  
73      152    21.86     4.80         27.3   7   3  
74      151    18.68     4.15         26.9   5   4  
75      151    31.90     4.70         40.6   4   1  
76      151    33.49     4.94         40.6   1   0  

[77 rows x 12 columns]

In [48]:
# splitting the 'Player' column to get the information about 'Country'
df_player = df['Player'].str.split("(", expand=True)

# concatinating 'Country' with the main dataframe
df = pd.concat([df, df_player], axis=1)

# dropping the 'Player' columns
df = df.drop('Player', axis=1)

# renaming the column names
df = df.rename(columns={0: 'Player',
                        1: 'Country'})

# remove the ")" from the 'Country' column
df['Country'] = df['Country'].str.replace(")", "")

# rearrange the columns
#new_col_sequence = ['Player', 'Span', 'Match', 'Innings', 'Balls', 'Runs', 'Wickets',
                    #'Average', 'Economy', 'Strike Rate',5,10, 'Country']
#df = df[new_col_sequence]

display(df.head())

C:\Users\Tcc\AppData\Local\Temp\ipykernel_6472\3550566742.py:15: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df['Country'] = df['Country'].str.replace(")", "")


Span  Match  Innings  Balls   Runs  Wickets  Average  Economy  \
0  1993-2011    350      341  18811  12326      534    23.08     3.93   
1  1984-2003    356      351  18186  11812      502    23.52     3.89   
2  1989-2003    262      258  12698   9919      416    23.84     4.68   
3  1994-2008    322      320  15775  11014      400    27.53     4.18   
4  1996-2015    398      372  17670  13632      395    34.51     4.62   

   Strike Rate   4   5           Player       Country  
0         35.2  15  10  M Muralitharan    Asia/ICC/SL  
1         36.2  17   6     Wasim Akram            PAK  
2         30.5  14  13    Waqar Younis            PAK  
3         39.4   9   4      WPUJC Vaas        Asia/SL  
4         44.7   4   9   Shahid Afridi   Asia/ICC/PAK

In [49]:
def icc_check(x):
    if "ICC" in x:
        return "Yes"
    else:
        return "No"

In [50]:
df['played_for_ICC'] = df['Country'].apply(icc_check)
display(df.head(10))
df['played_for_ICC'].value_counts()

Span  Match  Innings  Balls   Runs  Wickets  Average  Economy  \
0  1993-2011    350      341  18811  12326      534    23.08     3.93   
1  1984-2003    356      351  18186  11812      502    23.52     3.89   
2  1989-2003    262      258  12698   9919      416    23.84     4.68   
3  1994-2008    322      320  15775  11014      400    27.53     4.18   
4  1996-2015    398      372  17670  13632      395    34.51     4.62   
5  1996-2008    303      297  15712   9631      393    24.50     3.67   
6  1993-2007    250      248  12970   8391      381    22.02     3.88   
7  2000-2012    221      217  11185   8877      380    23.36     4.76   
8  2004-2019    226      220  10936   9760      338    28.87     5.35   
9  1990-2007    271      265  14496  10412      337    30.89     4.30   

   Strike Rate   4   5           Player       Country played_for_ICC  
0         35.2  15  10  M Muralitharan    Asia/ICC/SL            Yes  
1         36.2  17   6     Wasim Akram            PAK             No  
2         30.5  14  13    Waqar Younis            PAK             No  
3         39.4   9   4      WPUJC Vaas        Asia/SL             No  
4         44.7   4   9   Shahid Afridi   Asia/ICC/PAK            Yes  
5         39.9  12   5      SM Pollock     Afr/ICC/SA            Yes  
6         34.0   9   7      GD McGrath        AUS/ICC            Yes  
7         29.4  14   9           B Lee            AUS             No  
8         32.3  11   8      SL Malinga             SL             No  
9         43.0   8   2        A Kumble     Asia/INDIA             No

No     64
Yes    13
Name: played_for_ICC, dtype: int64

# Observation
1. 13 playes played for ICC

In [51]:
df['Country'] = df['Country'].str.replace("ICC/", "")

display(df.head())

Span  Match  Innings  Balls   Runs  Wickets  Average  Economy  \
0  1993-2011    350      341  18811  12326      534    23.08     3.93   
1  1984-2003    356      351  18186  11812      502    23.52     3.89   
2  1989-2003    262      258  12698   9919      416    23.84     4.68   
3  1994-2008    322      320  15775  11014      400    27.53     4.18   
4  1996-2015    398      372  17670  13632      395    34.51     4.62   

   Strike Rate   4   5           Player   Country played_for_ICC  
0         35.2  15  10  M Muralitharan    Asia/SL            Yes  
1         36.2  17   6     Wasim Akram        PAK             No  
2         30.5  14  13    Waqar Younis        PAK             No  
3         39.4   9   4      WPUJC Vaas    Asia/SL             No  
4         44.7   4   9   Shahid Afridi   Asia/PAK            Yes

In [52]:
# splitting the 'Span' column based on the "-"
df_span = df['Span'].str.split("-", expand=True)

# conccatinating the new dataframe with the main dataframe
df = pd.concat([df, df_span], axis=1)

# renaming the newly created column names
df = df.rename(columns={0: "start_year",
                        1: "end_year"})
df.head()

Span  Match  Innings  Balls   Runs  Wickets  Average  Economy  \
0  1993-2011    350      341  18811  12326      534    23.08     3.93   
1  1984-2003    356      351  18186  11812      502    23.52     3.89   
2  1989-2003    262      258  12698   9919      416    23.84     4.68   
3  1994-2008    322      320  15775  11014      400    27.53     4.18   
4  1996-2015    398      372  17670  13632      395    34.51     4.62   

   Strike Rate   4   5           Player   Country played_for_ICC start_year  \
0         35.2  15  10  M Muralitharan    Asia/SL            Yes       1993   
1         36.2  17   6     Wasim Akram        PAK             No       1984   
2         30.5  14  13    Waqar Younis        PAK             No       1989   
3         39.4   9   4      WPUJC Vaas    Asia/SL             No       1994   
4         44.7   4   9   Shahid Afridi   Asia/PAK            Yes       1996   

  end_year  
0     2011  
1     2003  
2     2003  
3     2008  
4     2015

In [53]:

df['start_year'] = df['start_year'].astype('int') 
df['end_year'] = df['end_year'].astype('int')

print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Span            77 non-null     object 
 1   Match           77 non-null     int64  
 2   Innings         77 non-null     int64  
 3   Balls           77 non-null     int64  
 4   Runs            77 non-null     int64  
 5   Wickets         77 non-null     int64  
 6   Average         77 non-null     float64
 7   Economy         77 non-null     float64
 8   Strike Rate     77 non-null     float64
 9   4               77 non-null     int64  
 10  5               77 non-null     int64  
 11  Player          77 non-null     object 
 12  Country         77 non-null     object 
 13  played_for_ICC  77 non-null     object 
 14  start_year      77 non-null     int32  
 15  end_year        77 non-null     int32  
dtypes: float64(3), int32(2), int64(7), object(4)
memory usage: 9.1+ KB
None


In [54]:
df['years_played'] = df['end_year'] - df['start_year']

df = df.drop(['start_year', "end_year"], axis=1)

display(df.head(10))

Span  Match  Innings  Balls   Runs  Wickets  Average  Economy  \
0  1993-2011    350      341  18811  12326      534    23.08     3.93   
1  1984-2003    356      351  18186  11812      502    23.52     3.89   
2  1989-2003    262      258  12698   9919      416    23.84     4.68   
3  1994-2008    322      320  15775  11014      400    27.53     4.18   
4  1996-2015    398      372  17670  13632      395    34.51     4.62   
5  1996-2008    303      297  15712   9631      393    24.50     3.67   
6  1993-2007    250      248  12970   8391      381    22.02     3.88   
7  2000-2012    221      217  11185   8877      380    23.36     4.76   
8  2004-2019    226      220  10936   9760      338    28.87     5.35   
9  1990-2007    271      265  14496  10412      337    30.89     4.30   

   Strike Rate   4   5           Player     Country played_for_ICC  \
0         35.2  15  10  M Muralitharan      Asia/SL            Yes   
1         36.2  17   6     Wasim Akram          PAK             No   
2         30.5  14  13    Waqar Younis          PAK             No   
3         39.4   9   4      WPUJC Vaas      Asia/SL             No   
4         44.7   4   9   Shahid Afridi     Asia/PAK            Yes   
5         39.9  12   5      SM Pollock       Afr/SA            Yes   
6         34.0   9   7      GD McGrath      AUS/ICC            Yes   
7         29.4  14   9           B Lee          AUS             No   
8         32.3  11   8      SL Malinga           SL             No   
9         43.0   8   2        A Kumble   Asia/INDIA             No   

   years_played  
0            18  
1            19  
2            14  
3            14  
4            19  
5            12  
6            14  
7            12  
8            15  
9            17

In [55]:
df.sort_values(by='years_played', ascending = False).head(10)

Span  Match  Innings  Balls   Runs  Wickets  Average  Economy  \
72  1989-2012    463      270   8054   6850      154    44.48     5.10   
10  1989-2011    445      368  14874  11871      323    36.75     4.78   
66  1999-2019    287      217   7958   6192      158    39.18     4.66   
62  1999-2019    301      199   7424   5926      167    35.48     4.78   
4   1996-2015    398      372  17670  13632      395    34.51     4.62   
20  2001-2020    220      220  10922   8893      270    32.93     4.88   
1   1984-2003    356      351  18186  11812      502    23.52     3.89   
0   1993-2011    350      341  18811  12326      534    23.08     3.93   
50  1974-1992    175      153   7461   4844      182    26.61     3.89   
12  1997-2015    295      277  14060   9674      305    31.71     4.12   

    Strike Rate   4   5             Player   Country played_for_ICC  \
72         52.2   4   2      SR Tendulkar      INDIA             No   
10         46.0   8   4     ST Jayasuriya    Asia/SL             No   
66         50.3   1   0      Shoaib Malik        PAK             No   
62         44.4   3   1          CH Gayle         WI            Yes   
4          44.7   4   9     Shahid Afridi   Asia/PAK            Yes   
20         40.4   7   1  Mashrafe Mortaza   Asia/BAN             No   
1          36.2  17   6       Wasim Akram        PAK             No   
0          35.2  15  10    M Muralitharan    Asia/SL            Yes   
50         40.9   3   1        Imran Khan        PAK             No   
12         46.0   8   2        DL Vettori         NZ            Yes   

    years_played  
72            23  
10            22  
66            20  
62            20  
4             19  
20            19  
1             19  
0             18  
50            18  
12            18

In [56]:
df.sort_values(by='years_played', ascending = True).head(10)

Span  Match  Innings  Balls  Runs  Wickets  Average  Economy  \
40  1994-2001    161      160   8129  6332      196    32.30     4.67   
74  2015-2022     80       76   4074  2821      151    18.68     4.15   
73  2008-2015     87       84   4154  3324      152    21.86     4.80   
49  2008-2015    113      112   6000  4182      184    22.72     4.18   
44  1996-2004    171      164   7336  5751      192    29.95     4.70   
56  2011-2019    107      104   5541  4297      173    24.83     4.65   
14  1995-2003    169      165   8770  6275      288    21.78     4.29   
55  2001-2009    116      116   5759  4240      174    24.36     4.41   
58  2004-2012    120      118   5855  5142      173    29.72     5.26   
15  1998-2007    191      188   9484  8021      288    27.85     5.07   

    Strike Rate   4  5            Player Country played_for_ICC  years_played  
40         41.4   3  1       BKV Prasad    INDIA             No             7  
74         26.9   5  4      Rashid Khan      AFG             No             7  
73         27.3   7  3       BAW Mendis       SL             No             7  
49         32.6   6  2      Saeed Ajmal      PAK             No             7  
44         38.2   1  6       L Klusener       SA             No             8  
56         32.0   7  3      Imran Tahir       SA             No             8  
14         30.4  11  6  Saqlain Mushtaq      PAK             No             8  
55         33.0   5  2       NW Bracken      AUS             No             8  
58         33.8   5  2        IK Pathan    INDIA             No             8  
15         32.9  10  2       AB Agarkar    INDIA             No             9

# Observation
1. SR Tendulkar from India played longest period of time
2. BKV Pasad from India played shortest period of time

In [57]:
df['Country'] = df['Country'].str.replace("ICC/", "")

display(df.head())

Span  Match  Innings  Balls   Runs  Wickets  Average  Economy  \
0  1993-2011    350      341  18811  12326      534    23.08     3.93   
1  1984-2003    356      351  18186  11812      502    23.52     3.89   
2  1989-2003    262      258  12698   9919      416    23.84     4.68   
3  1994-2008    322      320  15775  11014      400    27.53     4.18   
4  1996-2015    398      372  17670  13632      395    34.51     4.62   

   Strike Rate   4   5           Player   Country played_for_ICC  years_played  
0         35.2  15  10  M Muralitharan    Asia/SL            Yes            18  
1         36.2  17   6     Wasim Akram        PAK             No            19  
2         30.5  14  13    Waqar Younis        PAK             No            14  
3         39.4   9   4      WPUJC Vaas    Asia/SL             No            14  
4         44.7   4   9   Shahid Afridi   Asia/PAK            Yes            19

In [58]:
df['Country'].value_counts()

INDIA         9
PAK           9
NZ            8
AUS           8
WI            6
SL            5
ENG           4
SA            4
Asia/SL       4
Asia/INDIA    4
Afr/SA        4
Asia/PAK      3
BAN           2
AUS/ICC       2
ENG/ICC       2
Asia/BAN      1
Afr/ZIM       1
AFG           1
Name: Country, dtype: int64

# Observation
1. There are 10 countries in this dataset
2. There are 8 Australian bowler
3. There are 2 Bangladeshi bowler

In [59]:
df.sort_values(by='Economy', ascending = True).head(10)

Span  Match  Innings  Balls   Runs  Wickets  Average  Economy  \
65  1973-1990    115      112   6182   3407      158    21.56     3.30   
32  1988-2000    176      175   9353   5429      225    24.12     3.48   
67  1980-1992    136      134   7175   4233      157    26.96     3.53   
5   1996-2008    303      297  15712   9631      393    24.50     3.67   
25  1978-1994    225      221  11202   6945      253    27.45     3.71   
31  1985-2000    205      204  10822   6918      227    30.47     3.83   
6   1993-2007    250      248  12970   8391      381    22.02     3.88   
1   1984-2003    356      351  18186  11812      502    23.52     3.89   
50  1974-1992    175      153   7461   4844      182    26.61     3.89   
0   1993-2011    350      341  18811  12326      534    23.08     3.93   

    Strike Rate   4   5           Player  Country played_for_ICC  years_played  
65         39.1   1   5       RJ Hadlee        NZ             No            17  
32         41.5   6   4     CEL Ambrose        WI             No            12  
67         45.7   6   0     MD Marshall        WI             No            12  
5          39.9  12   5      SM Pollock    Afr/SA            Yes            12  
25         44.2   3   1     N Kapil Dev     INDIA             No            16  
31         47.6   6   1        CA Walsh        WI             No            15  
6          34.0   9   7      GD McGrath   AUS/ICC            Yes            14  
1          36.2  17   6     Wasim Akram       PAK             No            19  
50         40.9   3   1      Imran Khan       PAK             No            18  
0          35.2  15  10  M Muralitharan   Asia/SL            Yes            18

In [60]:
df.sort_values(by='Strike Rate', ascending = True).head(10)

Span  Match  Innings  Balls  Runs  Wickets  Average  Economy  \
41  2010-2021     99       99   5099  4379      195    22.45     5.15   
74  2015-2022     80       76   4074  2821      151    18.68     4.15   
73  2008-2015     87       84   4154  3324      152    21.86     4.80   
7   2000-2012    221      217  11185  8877      380    23.36     4.76   
60  2012-2021     93       93   5117  4261      169    25.21     4.99   
14  1995-2003    169      165   8770  6275      288    21.78     4.29   
2   1989-2003    262      258  12698  9919      416    23.84     4.68   
46  2007-2018    117      114   5760  4761      188    25.32     4.95   
28  2005-2015    153      150   7489  6038      239    25.26     4.83   
26  1998-2011    163      162   7764  6169      247    24.97     4.76   

    Strike Rate   4   5            Player   Country played_for_ICC  \
41         26.1  11   8         MA Starc        AUS             No   
74         26.9   5   4      Rashid Khan        AFG             No   
73         27.3   7   3       BAW Mendis         SL             No   
7          29.4  14   9            B Lee        AUS             No   
60         30.2   8   5         TA Boult         NZ             No   
14         30.4  11   6  Saqlain Mushtaq        PAK             No   
2          30.5  14  13     Waqar Younis        PAK             No   
46         30.6   7   2         M Morkel     Afr/SA             No   
28         31.3   9   3       MG Johnson        AUS             No   
26         31.4   6   4    Shoaib Akhtar   Asia/PAK            Yes   

    years_played  
41            11  
74             7  
73             7  
7             12  
60             9  
14             8  
2             14  
46            11  
28            10  
26            13

In [61]:
df.sort_values(by='Average', ascending = True).head(10)

Span  Match  Innings  Balls   Runs  Wickets  Average  Economy  \
74  2015-2022     80       76   4074   2821      151    18.68     4.15   
65  1973-1990    115      112   6182   3407      158    21.56     3.30   
19  1991-2003    164      162   8561   5926      272    21.78     4.15   
14  1995-2003    169      165   8770   6275      288    21.78     4.29   
73  2008-2015     87       84   4154   3324      152    21.86     4.80   
6   1993-2007    250      248  12970   8391      381    22.02     3.88   
41  2010-2021     99       99   5099   4379      195    22.45     5.15   
49  2008-2015    113      112   6000   4182      184    22.72     4.18   
0   1993-2011    350      341  18811  12326      534    23.08     3.93   
7   2000-2012    221      217  11185   8877      380    23.36     4.76   

    Strike Rate   4   5            Player  Country played_for_ICC  \
74         26.9   5   4      Rashid Khan       AFG             No   
65         39.1   1   5        RJ Hadlee        NZ             No   
19         31.4  11   2        AA Donald        SA             No   
14         30.4  11   6  Saqlain Mushtaq       PAK             No   
73         27.3   7   3       BAW Mendis        SL             No   
6          34.0   9   7       GD McGrath   AUS/ICC            Yes   
41         26.1  11   8         MA Starc       AUS             No   
49         32.6   6   2      Saeed Ajmal       PAK             No   
0          35.2  15  10   M Muralitharan   Asia/SL            Yes   
7          29.4  14   9            B Lee       AUS             No   

    years_played  
74             7  
65            17  
19            12  
14             8  
73             7  
6             14  
41            11  
49             7  
0             18  
7             12

# Observation
1. RJ Hadlee from New Zeland have the lowest economy rate of 3.30
2. MA Starc from Australia have the highest economy rate of 5.15
3. Rashid Khan from Afganistan have the lowest aveage of 18.68